### PR TITLE
Bean post processor

### PR DESCRIPTION
**Pull request by VolodymyrM27**

### CODE REVIEW

File: src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneCustomMongoRepository.kt [hunk 1]
```diff
@@ -30,4 +31,14 @@ class DroneCustomMongoRepository(
...
         mongoTemplate.updateFirst(query, update, Drone::class.java)
     }
+
+    override fun findAllDronesWhereTheRemainingBatteryChargeIsLessThanAndHaveTheStatuses(
+        batteryLevel: Double,
+        statusesList: List<DroneStatus>
+    ): List<Drone> {
+        val query = Query()
+            .addCriteria(Criteria.where("batteryLevel").lte(batteryLevel))
+            .addCriteria(Criteria.where("status").`in`(statusesList))
```
Comment (ajax-gnatiuk-s) [inline src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneCustomMongoRepository.kt:41]: I don't force you changing this code, just sharing alternative approach:
```kotlin
val criteria  = Criteria.where("batteryLevel").lte(batteryLevel)
                              .and("status").`in`(statusesList)
val query = Query(criteria)
```

File: src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneCustomRepository.kt [hunk 2]
```diff
@@ -1,7 +1,12 @@
 package systems.ajax.motrechko.airguardian.repository
 
+import systems.ajax.motrechko.airguardian.enums.DroneStatus
 import systems.ajax.motrechko.airguardian.model.Drone
 
 interface DroneCustomRepository {
     fun updateDroneInfo(drone: Drone)
+    fun findAllDronesWhereTheRemainingBatteryChargeIsLessThanAndHaveTheStatuses(
```
Comment (ajax-gnatiuk-s) [inline src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneCustomRepository.kt:8]:
```suggestion
    
    fun findAllDronesWhereTheRemainingBatteryChargeIsLessThanAndHaveTheStatuses(
```

**Review by ajax-gnatiuk-s — CHANGES_REQUESTED** [review]

File: src/main/kotlin/systems/ajax/motrechko/airguardian/model/BatteryApplication.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package systems.ajax.motrechko.airguardian.model
+
+data class BatteryApplication(
+    var serviceMessage: String = "",
+    var drone: Drone
```
[the remark this answers is not in the document]
Comment (VolodymyrM27) [reply src/main/kotlin/systems/ajax/motrechko/airguardian/model/BatteryApplication.kt:5]: I needed it for something, but I've already cut out this functionality

File: src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneCustomMongoRepository.kt (hunk 1 above)
Comment (VolodymyrM27) [reply src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneCustomMongoRepository.kt:41]: Thanks, I was interesting which of the options is better to use?

File: src/main/kotlin/systems/ajax/motrechko/airguardian/beanPostProcessor/MyScheduledBeanPostProcessor.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.stereotype.Component
+import java.util.*
+import kotlin.concurrent.fixedRateTimer
+
+@Component
+class MyScheduledBeanPostProcessor : BeanPostProcessor {
+    override fun postProcessAfterInitialization(bean: Any, beanName: String): Any? {
+        val beanClass = bean.javaClass
```
[the remark this answers is not in the document]
Comment (VolodymyrM27) [reply src/main/kotlin/systems/ajax/motrechko/airguardian/beanPostProcessor/MyScheduledBeanPostProcessor.kt:11]: I don't quite understand the question. Shouldn't the Validated annotation be on the controller, not on the service ?

File: src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneCustomMongoRepository.kt (hunk 1 above)
Comment (ajax-gnatiuk-s) [reply src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneCustomMongoRepository.kt:41]: your approach can be used to build some more complex queries. My approach is less verbose but less flexible at the same time

File: src/main/kotlin/systems/ajax/motrechko/airguardian/beanPostProcessor/MyScheduledBeanPostProcessor.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (ajax-gnatiuk-s) [reply src/main/kotlin/systems/ajax/motrechko/airguardian/beanPostProcessor/MyScheduledBeanPostProcessor.kt:11]: Ok, it's just not a question itself. 

Let me rephrase. You have the service `DroneBatteryCheckService` that has function annotated with `MyScheduled`.
Just add to `DroneBatteryCheckService` annotation `@Validated`.
```kotlin
@Validated // <----------------------- Just add this annotation
@Service
class DroneBatteryCheckService(
...
) {
    @MyScheduled(delay = 1000, period = 5000)
    fun checkTheBatteriesOfAllDrones() {
    ....
    }
}
```

And check that your bean post processor still works.

----

I see an issue with your BPP that's why I ask you to experiment with your code

----

**NOTE:** my point is not about adding `@Validated` to your service. It's just an option to verify that BPP works or doesn't work.
